pkg/broker/ingress/handler.go

```diff
@@ -167,3 +175,16 @@ func (h *handler) toEvent(request *nethttp.Request) (event *cev2.Event, msg stri
 	}
 	return event, "", nethttp.StatusOK
 }
+
+func (h *handler) reportMetrics(ns, broker string, event *cev2.Event, start time.Time, statusCode int) {
+	eventType := "Unknown"
+	if event != nil {
+		eventType = event.Type()
+	}
+	ctxWithTag, err := generateTag(h.ctx, ns, broker, eventType, statusCode)
```

Comment:
I would apply most tag mutators at the beginning of each request and simply use the request ctx + additional tags to report metrics. I would also inline the generateTag helper, since this helper has too many string type arguments and I think directly manipulating the tags is a little more readable even if it is more verbose.

Reply:
Consolidated to a single function in StatsReporter, which also takes in the request context.